src/app/content/utils.ts

```diff
@@ -26,18 +27,22 @@ export interface ContentPageRefencesType {
   pageId: string;
 }
 
-export function getContentPageReferences(htmlContent: string) {
-  const matches: ContentPageRefencesType[] = (htmlContent.match(/.\/([a-z0-9-]+(@[\d.]+)?):([a-z0-9-]+.xhtml)/g) || [])
+export function getContentPageReferences(book: ArchiveBook, page: ArchivePage) {
+  const matches: ContentPageRefencesType[] = (
+    page.content.match(/['"]{1}((#[^'"\s]+)|(\.\/([a-z0-9-]+(@[\d.]+)?):([a-z0-9-]+.xhtml(#.[^'"]+)?)))/g) || []
+    )
     .map((match) => {
       const [bookMatch, pageMatch] = match.split(':');
-      const pageId = pageMatch.split('.xhtml')[0];
-      const [bookId, bookVersion] = bookMatch.split('@') as [string, string | undefined];
+      const pageId = pageMatch && pageMatch.split('.xhtml')[0];
+      const [bookId, bookVersion] = bookMatch && bookMatch.split('@') as [string, string | undefined];
+
+      console.log('book: ', book, book.id);
 
       return {
-        bookId: bookId.substr(2),
-        bookVersion,
-        match,
-        pageId: stripIdVersion(pageId),
+        bookId: (match.includes(':') && bookId && bookId.substr(3)) || book.id,
```

Comment:
@TomWoodward Everything was passing `bookId && bookId.substr(3)` so I added the first condition, but is there a better way to determine that we have what we need from the match?

Reply:
yea i mean the better way would be to use regex groups, but this is fine